deploy/cephcsi/image/Dockerfile

```diff
@@ -1,15 +1,50 @@
+ARG SRC_DIR="/go/src/github.com/ceph/ceph-csi/"
+
 FROM ceph/ceph:v15
-LABEL maintainers="Ceph-CSI Authors"
-LABEL description="Ceph-CSI Plugin"
 
-# To support cross building, do NOT RUN native binaries here.
-# If we have to run native binaries, qemu-user-static must be installed on
-# build host and mounted to container.
+LABEL stage="build"
+
+ARG GOLANG_VERSION=1.13.9
+ARG CSI_IMAGE_NAME=quay.io/cephcsi/cephcsi
+ARG CSI_IMAGE_VERSION=canary
+ARG ARCH
+ARG GIT_COMMIT
+
+RUN dnf install libcephfs-devel librados-devel librbd-devel /usr/bin/cc /usr/bin/c++ make -y
+
+RUN mkdir -p /usr/local/go && \
+    curl https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-${ARCH}.tar.gz | tar xzf - -C /usr/local/go --strip-components=1
+
+
+ENV GOROOT=/usr/local/go \
+    GOPATH=/go \
+    PATH=$GOROOT/bin:$GOPATH/bin:$PATH\
+    CGO_ENABLED=1\
+    GIT_COMMIT="${GIT_COMMIT}"\
+    ENV_CSI_IMAGE_VERSION="${CSI_IMAGE_VERSION}"\
+    ENV_CSI_IMAGE_NAME="${CSI_IMAGE_NAME}"\
+    PATH="${GOROOT}/bin:${GOPATH}/bin:${PATH}"
+
+
+WORKDIR ${SRC_DIR}
+
+# Copy source directories
+COPY . ${SRC_DIR}
+
+RUN ls -lrt
+# Build executable
+RUN make cephcsi
+
+RUN _output/cephcsi --version
+
+#-- Final container
+FROM ceph/ceph:v15
 
-# Removing ceph-iscsi repo to workaround the repo issue while upgrading
-#RUN rm -f /etc/yum.repos.d/ceph-iscsi.repo && yum -y update && yum clean all
-ENV CSIBIN=/usr/local/bin/cephcsi
+LABEL maintainers="Ceph-CSI Authors"\
+    version=$ENV_CSI_IMAGE_VERSION\
+    architecture=$GOARCH\
+    description="Ceph-CSI Plugin"
 
-COPY cephcsi $CSIBIN
+COPY --from=0 ${SRC_DIR}/_output/cephcsi /usr/local/bin/cephcsi
```

Comment:
instead of using `--from=0`, add `AS builder` to the first `FROM` line and use `--from=builder`. It is a little more verbose, but also more obvious what is done.

Reply:
yeah thought of doing it. we need more changes in deploy.sh and build-multi-arch.sh for this one as we are replacing the FROM image name during the multi-stage build. this can be done as an enhancement